src/config.js

```diff
@@ -454,6 +454,13 @@ function parseArgs(options, raw_args) {
 
     args.concurrency = computeConcurrency(args.concurrency);
 
+    // Watch mode is typically used for local development. In that case
+    // we need to keep some cores available for other stuff like
+    // a bundler and/or unit test runner for frontend developers.
+    if (args.watch && args.concurrency > getCPUCount()) {
+        args.concurrency = Math.max(1, getCPUCount() - 2);
```

Comment:
Wouldn't this break if somebody manually sets the concurrency? Can we change the default so that it varies depending on whether `-w` is set?

Reply:
Good call, I've updated the PR so that it only affects the default value. Let me know what you think.